de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/impl/BPartnerDAO.java

```diff
@@ -1042,10 +1042,10 @@ public BPartnerLocationId retrieveBPartnerLocationId(@NonNull final BPartnerLoca
 		final I_C_BPartner_Location ownToLocation = queryBuilder
 				.create()
 				.first();
-		if (!(query.isAlsoTryRelation() && query.getRelationBPartnerLocationId() != null) && ownToLocation != null)
+
+		// if not matching location is set and we found already a location, use that one
+		if (query.getRelationBPartnerLocationId() == null && ownToLocation != null)
```

Comment:
* from looking at the code below, *I think* that 
if `query.getRelationBPartnerLocationId() ==null`, 
then we always return `ownToLocation` (also if it's null), right?

If that's the case, then pls return `ownToLocation` already up here, if  `query.getRelationBPartnerLocationId() ==null`. 
That makes it easier to understand the code.